src/Compilers/VisualBasic/Portable/BoundTree/Expression.vb

```diff
@@ -126,7 +140,7 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
             Dim rightOperatorBinary As BoundUserDefinedBinaryOperator = TryCast(Me.Right, BoundUserDefinedBinaryOperator)
             If rightOperatorBinary IsNot Nothing Then
-                If TypeOf rightOperatorBinary.Left Is BoundCompoundAssignmentTargetPlaceholder Then
+                If TypeOf DirectCast(rightOperatorBinary, IBinaryOperatorExpression).LeftOperand Is BoundCompoundAssignmentTargetPlaceholder Then
```

Comment:
This is possibly better than before, but the whole logic in this method is too complex and is incorrect in general. I opened a special issue regarding this.